internal/app/machined/pkg/controllers/block/internal/volumes/format.go

```diff
@@ -30,7 +30,7 @@ func Format(ctx context.Context, logger *zap.Logger, volumeContext ManagerContex
 		devPath = volumeContext.Status.Location
 	}
 
-	dev, err := blockdev.NewFromPath(devPath, blockdev.OpenForWrite())
+	dev, err := blockdev.NewFromPath(devPath)
```

Comment:
dropped write bit here, as it's not needed, and breaks when the media is read-only (we're not writing to it anyways in this function)